pyatlan/model/typedef.py

```diff
@@ -1,241 +1,462 @@
+        description="Username of the user who last assets_updated the object."
     )
+    version: Optional[int] = Field(description="Version of this type definition.")
 
 
 class EnumDef(TypeDef):
     class ElementDef(AtlanObject):
-        value: str = Field(None, description="Unused.")
-        description: Optional[str] = Field(None, description="Unused.")
-        ordinal: Optional[int] = Field(None, description="Unused.")
+        value: str = Field(description="One unique value within the enumeration.")
+        description: Optional[str] = Field(description="Unused.")
+        ordinal: Optional[int] = Field(
+            description="Unique numeric identifier for the value."
+        )
+
+        @staticmethod
+        def of(ordinal: int, value: str) -> EnumDef.ElementDef:
+            from pyatlan.model.assets import validate_required_fields
+
+            validate_required_fields(
+                ["ordinal", "value"],
+                [ordinal, value],
+            )
+            return EnumDef.ElementDef(ordinal=ordinal, value=value)
+
+        @staticmethod
+        def list_from(values: List[str]) -> List[EnumDef.ElementDef]:
+            from pyatlan.model.assets import validate_required_fields
+
+            validate_required_fields(
+                ["values"],
+                [values],
+            )
+            elements = []
+            if values:
+                for i in range(0, len(values), 1):
+                    elements.append(EnumDef.ElementDef.of(ordinal=i, value=values[i]))
+            return elements
 
     category: AtlanTypeCategory = AtlanTypeCategory.ENUM
-    element_defs: List["EnumDef.ElementDef"] = Field(None, description="Unused.")
-    options: Optional[Dict[str, Any]] = Field(
-        None, description="Optional properties of the type_ definition."
+    element_defs: List["EnumDef.ElementDef"] = Field(
+        description="Valid values for the enumeration."
     )
-    service_type: Optional[str] = Field(
-        None, description="Internal use only.", example="atlan"
+    options: Optional[Dict[str, Any]] = Field(
+        description="Optional properties of the type definition."
     )
+    service_type: Optional[str] = Field(description="Internal use only.")
+
+    @staticmethod
+    def create(name: str, values: List[str]) -> EnumDef:
+        from pyatlan.model.assets import validate_required_fields
+
+        validate_required_fields(
+            ["name", "values"],
+            [name, values],
+        )
+        # Explicitly set all defaults to ensure inclusion during pydantic serialization
+        return EnumDef(
+            category=AtlanTypeCategory.ENUM,
+            name=name,
+            element_defs=EnumDef.ElementDef.list_from(values),
+        )
+
+    def get_valid_values(self) -> Optional[List[str]]:
+        """
+        Translate the element definitions in this enumeration into simple list of strings.
+        """
+        if self.element_defs:
```

Comment:
I would suggest replacing lines 228/234 with a list comprehension as follows:
`return [one.value for one in self.element_defs] if self.element_defs else []`